### PR TITLE
feat: ui improvements

### PR DESCRIPTION
**Pull request by 0xpatrickdev**
- [refactor: split into components](https://github.com/Agoric/dapp-offer-up/pull/25/commits/a641d5298daf6f1a8efdbb882f745024e1b52786)
- [feat(ui): improve styling](https://github.com/Agoric/dapp-offer-up/pull/25/commits/ed60c8792cec38a23acc1023102b1c8e85c45ba8)

- #63 should be merged in first

Still room for improvement, ideally the user shouldn't have to scroll, but I think this looks a little bit better:

Screenshot 2023-12-29 at 4 51 11 PM
Screenshot 2023-12-29 at 4 50 54 PM

### CODE REVIEW

File: ui/src/components/Trade.tsx [hunk 1]
```diff
@@ -0,0 +1,141 @@
+import { FormEvent, useState } from 'react';
+import { stringifyAmountValue } from '@agoric/ui-components';
+import scrollIcon from '../assets/scroll.png';
+import istIcon from '../assets/IST.png';
```
Comment (dckc) [inline ui/src/components/Trade.tsx:4]: Do we have `.svg` handy?

File: ui/src/components/Trade.tsx [hunk 2]
```diff
@@ -0,0 +1,141 @@
...
+            />
+          ))}
+        </div>
+      </div>
+      <div className="trade">
+        <h3>Give: Offer at least 0.25 IST</h3>
+        <div className="row-center">
+          <Item
+            key="IST"
+            coinIcon={istIcon}
```
Comment (dckc) [inline ui/src/components/Trade.tsx:111]: My eye very much expects fungibles to be written horizontally: 10 USD. 5 IST.
stacking IST on top of the logo seems awkward.

image

**Review by dckc — APPROVED** [review]
OK by me. Let's try to show it to product before merging, though.

it feels a little too tall, but I don't have an alternative to suggest

**Comment by dckc** [comment]
> ideally the user shouldn't have to scroll

I'd like to see how it looks with the give / want side by side. Is it obvious to you how to do that?

**Review by samsiegart — APPROVED** [review]
Approving because it's an improvement. Similar concerns as Dan, we can consult product about whether it's worth creating an issue for a more complete redesign.

**Comment by dckc** [comment]
> Approving because it's an improvement.

Changing the UI makes it inconsistent with the docs, including the video walkthru. So it's not clear that we want to land lots of little improvements - do we update the docs and video each time? Having product make a conscious decision about whether this is the next update seems cost-effective.

cc @sufyaankhan @kbennett2000

**Comment by dckc** [comment]
> ... Having product make a conscious decision about whether this is the next update seems cost-effective.

@sufyaankhan says go.

File: ui/src/components/Trade.tsx (hunk 1 above)
Comment (0xpatrickdev) [reply ui/src/components/Trade.tsx]: Updated to use svg

File: ui/src/components/Trade.tsx (hunk 2 above)
Comment (0xpatrickdev) [reply ui/src/components/Trade.tsx:111]: Agree, maybe a ticket to use `<NatInput />` is worthwhile. But I'm not sure if that comes with styling.

**Review by samsiegart — APPROVED** [review]
LGTM

**Pull request merged**